overlord/servicestate/conflict.go

```diff
@@ -0,0 +1,94 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package servicestate
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/overlord/state"
+)
+
+// An AffectedQuotasFunc returns a list of affected quota group names for the given supported task.
+type AffectedQuotasFunc func(*state.Task) ([]string, error)
+
+var (
+	affectedQuotasByKind = make(map[string]AffectedQuotasFunc)
+)
+
+// AddAffectedQuotasByKind registers an AffectedQuotasFunc for returning the affected quotas
+// for tasks of the given kind, to use in conflicts detection.
+func AddAffectedQuotasByKind(kind string, f AffectedQuotasFunc) {
```

Comment:
Samuele already replied to my nitpick in https://github.com/snapcore/snapd/pull/11940#discussion_r952716679, but since this is a new function, I hope we can rename it. For the record, I'm not calling for total consistency: we already have quite a few registrations of handlers and callback functions that do not use `Func` in their name, such as `AddCleanup` and `AddCommand`, but in those cases it's rather obvious that they need to take a callback as parameter, so they are fine. And I'm not even insisting that "`Func`" should be part of the name, but I'm suggesting that we rename this function in a way that explains it a bit better, because right now when I read this function name I think "this function adds affected quotas by kind", which is rather misleading.

Grepping in the snapd codebase, I see that we have a `RegisterDeviceMapperBackResolver` which even without having `Func` in the name, makes it rather obvious that it takes a function. So, taking inspiration from that, maybe `RegisterAffectedQuotasEnumerator()` can be an option?

Reply:
the issue is that we have already other AddAffected* functions, I'm not against using a different pattern for their names, I'm against just now renaming this one.

Reply:
the other relevant functions are:

./overlord/snapstate/conflict.go:func AddAffectedSnapsByAttr(attr string, f AffectedSnapsFunc) {
./overlord/snapstate/conflict.go:func AddAffectedSnapsByKind(kind string, f AffectedSnapsFunc) {

Reply:
Ah, I see! Ok, then :-)